Source/Core/Core/PowerPC/Jit64/JitAsm.cpp

```diff
@@ -58,7 +58,8 @@ void Jit64AsmRoutineManager::Generate()
   MOV(64, PPCSTATE(stored_stack_pointer), R(RSP));
 
   // something that can't pass the BLR test
-  MOV(64, MDisp(RSP, 8), Imm32((u32)-1));
+  MOV(64, R(RSCRATCH2), Imm64(0xFFFF'FFFF'FFFF'FFFF));
+  MOV(64, MDisp(RSP, 8), R(RSCRATCH2));
```

Comment:
Is it important that the value exists in `RSCRATCH2`? I'm guessing it is, but it's not immediately obvious to me why.

Reply:
No. x86 just doesn't let you write a 64-bit immediate to memory in one instruction.

Reply:
https://godbolt.org/z/sz4zhT3nf

True, not any arbitrary 64-bit immediate, but for 64-bit instructions that encode a 32-bit immediate x86 sign extends the immediate. So if the most significant 33 bits are all ones or zeroes, you can do it in one instruction.

I think the code before this diff already used this.

Reply:
> I think the code before this diff already used this.

Oh yes, you're right. Somehow I thought the first parameter was 32 and not 64.